src/gui/guiConfirmRegistration.cpp

```diff
@@ -96,8 +96,34 @@ void GUIConfirmRegistration::regenerateGui(v2u32 screensize)
 				"Join to confirm account creation or click Cancel to "
 				"abort.");
 		char info_text_buf[1024];
+
+#ifdef _MSC_VER
+		int sizeBuffer = sizeof(info_text_buf) -
+				 (info_text_template.length() - 8 + address.length() +
+						 m_playername.length());
+
+		if (sizeBuffer >= 0) {
+			sprintf(info_text_buf, info_text_template.c_str(),
+					address.c_str(), m_playername.c_str());
+		} else {
+			warningstream << "Failed to generate info_text. Buffer is too "
+					 "small"
+				      << std::endl;
+			int new_size_text_template =
+					(int)info_text_template.size() + (sizeBuffer - 4);
+			if (new_size_text_template > 0) {
+				std::string info_text_template_trunc =
+						info_text_template.substr(0,
+								new_size_text_template);
+				info_text_template_trunc.append("...");
```

Comment:
instead of copying the whole data, just replace the last 3 chars directly (info_text_template[info_text_template.size() - 4] = '.')